torch/lib/c10d/test/ProcessGroupNCCLTest.cpp

```diff
@@ -386,6 +403,45 @@ void testAllgather(const std::string& path, int rank, int size) {
   std::cout << "Allgather test successful" << std::endl;
 }
 
+void testAllgatherWithTimeout(const std::string& path, int rank, int size) {
+  // First we must set NCCL_BLOCKING_WAIT to 1 for timeouts to work in
+  // ProcessGroupNCCL.
+  auto originalBlockingWait = getenv(c10d::NCCL_BLOCKING_WAIT);
+  setenv(c10d::NCCL_BLOCKING_WAIT, "1", 1);
+  auto test = AllgatherNCCLTestWithTimeout(path, size);
+  test.initialize(rank, size);
+  auto work = test.run();
+
+  // Wait for work to finish
+  try {
+    test.wait(work, kWorkTimeout);
+  } catch (const std::runtime_error& e) {
+    // First we reset the NCCL_BLOCKING_WAIT environment variable to prevent
+    // any unwanted side-effects. We must do this at all exit-points for this
+    // function.
+    setenv(c10d::NCCL_BLOCKING_WAIT, originalBlockingWait, 1);
```

Comment:
nit (please feel free to ignore): does it worth it to make this a guard?

Reply:
Good point. It's kind of overhead to restore original env value